hatch.toml

```diff
@@ -817,4 +817,48 @@ tested_pytest_plugin_version = ["v1", "v2"]
 matrix.tested_pytest_plugin_version.env-vars = [
     { key = "_TESTED_PYTEST_LEGACY_PLUGIN", value = "true", if = ["v1"]},
     { key = "_TESTED_PYTEST_LEGACY_PLUGIN", value = "false", if = ["v2"]}
-]
\ No newline at end of file
+]
+
+[envs.appsec_integrations_pygoat]
+template = "appsec_integrations_pygoat"
+dependencies = [
+    "pytest",
+    "pytest-cov",
+    "requests",
+    "hypothesis",
+    "pyyaml==6.0.1",
+]
+
+[envs.appsec_integrations_pygoat.env-vars]
+DD_CIVISIBILITY_ITR_ENABLED = "0"
+DD_IAST_REQUEST_SAMPLING = "100"
+DD_IAST_ENABLED = "true"
+_DD_IAST_DEBUG = "false"
+DD_IAST_VULNERABILITIES_PER_REQUEST = "100"
+DD_REMOTE_CONFIGURATION_ENABLED = "true"
+DD_IAST_DEDUPLICATION_ENABLED = "false"
+DD_TRACE_AGENT_URL = "http://0.0.0.0:9126"
+DD_FAST_BUILD = "1"
+PYTHONDONTWRITEBYTECODE = "1"
+PYTHONUNBUFFERED = "1"
+
+[envs.appsec_integrations_pygoat.scripts]
+test = [
+    "uname -a",
+    "pip freeze",
+    "rm -rf pygoat",
+    "tar -xf tests/appsec/integrations/pygoat_tests/fixtures/pygoat.xz -C .",
+    "pip install -r pygoat/requirements.txt",
+    "pip install --no-cache-dir --force-reinstall pyyaml==6.0.1 --global-option='--without-libyaml'",
+    "python3 pygoat/manage.py migrate",
+    "python3 pygoat/manage.py loaddata tests/appsec/integrations/pygoat_tests/fixtures/*.json",
+    "python -m ddtrace.commands.ddtrace_run python3 pygoat/manage.py runserver 0.0.0.0:8321 > /dev/null 2>&1 & echo $! > pygoat.pid",
+    "sleep 5",
+    "python -m pytest -vvv {args:tests/appsec/integrations/pygoat_tests/}",
+    "kill $(cat pygoat.pid) || true",
+    "rm -f pygoat.pid",
+    "rm -rf pygoat",
```

Comment:
there are many steps, could you move all of them to a "test/appsec/integrations/pygoat_tests/run.sh"?

Reply:
I don't really see the difference but done!